src/routers/posts.ts

```diff
@@ -22,25 +22,1077 @@ import {
+ *   put:
+ *     tags:
+ *       - Posts operations
+ *     summary: Change the range of a post , Must be connected as user
+ *     description: Updates the range of a post specified by its ID. Only admins can update any post, while regular users can only update posts they created. Superadmin posts cannot have their range changed.
+ *     security:
+ *       - BearerAuth: []  # Indicates that the endpoint requires a bearer token for authentication
+ *     parameters:
+ *       - in: path
+ *         name: postID
+ *         required: true
+ *         schema:
+ *           type: string
+ *         description: The unique identifier of the post whose range is to be updated.
+ *     requestBody:
+ *       description: The new range value to be set for the post.
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             properties:
+ *               range:
+ *                 type: string
+ *                 description: The new range to be set for the post.
+ *     responses:
+ *       200:
+ *         description: Success message with the result of the update operation.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 data:
+ *                   type: object
+ *                   description: The result of the update operation.
+ *                   properties:
+ *                     matchedCount:
+ *                       type: integer
+ *                       description: Number of documents matched for the update.
+ *                     modifiedCount:
+ *                       type: integer
+ *                       description: Number of documents modified.
+ *       401:
+ *         description: Unauthorized access error if trying to change the range of a superadmin post or if the user does not have permission to update the post.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 msg:
+ *                   type: string
+ *                   description: Error message indicating why the range could not be changed.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 msg:
+ *                   type: string
+ *                   description: Error message indicating that an error occurred.
+ *                 err:
+ *                   type: object
+ *                   description: Details of the error encountered.
+ */
 router.patch('/changeRange/:postID', auth, postRangeControl, changePostRange);
+
+/**
+ * @swagger
+ * /posts/changeActive/{postID}:
+ *   patch:
+ *     tags:
+ *       - Posts operations
+ *     summary: Toggle the active status of a post, Must be connected as admin
```

Comment:
summary should contains just -
"Must be connected as admin" OR
"Must be connected as user"